src/main/java/org/example/gabartischolarsqbbe/service/QuestionService.java

```diff
@@ -0,0 +1,195 @@
+package org.example.gabartischolarsqbbe.service;
+
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
+import org.example.gabartischolarsqbbe.dto.CsvUploadResponse;
+import org.example.gabartischolarsqbbe.entity.Question;
+import org.example.gabartischolarsqbbe.repository.QuestionRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class QuestionService {
+    
+    private final QuestionRepository questionRepository;
+    private final JobService jobService;
+    
+    @Autowired
+    public QuestionService(QuestionRepository questionRepository, JobService jobService) {
+        this.questionRepository = questionRepository;
+        this.jobService = jobService;
+    }
+    
+    /**
+     * Process CSV file and save questions to database
+     * CSV format: question_text,difficulty,job_title_id,option1,option1_correct,option2,option2_correct,option3,option3_correct,option4,option4_correct
+     * @param file the CSV file to process
+     * @return CsvUploadResponse with processing results
+     */
+    public CsvUploadResponse processCsvFile(MultipartFile file) {
```

Comment:
Rather than adding this here, it would be better to create an `util` called something like `QuestionsCSVImporter` and have this util handles the logic of extracting the fields and creating `List<Questions>`, and then create a method in the `QuestionService` that takes `List<Question>` and bulk creates the questions.


This is mainly a better approach for concern separation.
Thinks of it this way, the `QuestionService` is concerned with dealing with Question entities itself, at this point the `csv` file is yet not considered as a `Question` per se